include/tvm/runtime/ndarray.h

```diff
@@ -306,6 +324,26 @@ inline void NDArray::CopyFrom(const NDArray& other) {
   CopyFromTo(&(other.get_mutable()->dl_tensor), &(get_mutable()->dl_tensor));
 }
 
+inline void NDArray::CopyFromBytes(const void* data, size_t nbytes) {
```

Comment:
To avoid duplication, let us make it a non-line member function, and make uses of implementation in the C API here https://github.com/apache/incubator-tvm/blob/master/src/runtime/ndarray.cc#L289 

After that, we can safely redirect the C API calls into these C++ APIs

Reply:
Ok, I have it calling the TVMArrayCopyFrom/ToBytes methods.

Are you saying to put the implementation into ndarray.cc ?

Reply:
Ok moved CopyTo/FromBytes to ndarray.cc